### PR TITLE
Sr/cfto/sanity

### PR DESCRIPTION
**Pull request by sumitroy2011**
Please review both the commits to the feature branch and merge it with master

### CODE REVIEW

**Review by sumitroy2011 — COMMENTED** [review]
Sir i have made the changes which you've asked for.
Now kindly merge the changes to master

**Comment by sumitroy2011** [comment]
Please merge the incorporated changes to master

File: spec/all.globalBeforeEach.spec.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+  describe('Testing - CFTO UK Site', function() {
+
+
+function log_in()
```
Comment (SunilSyal) [inline spec/all.globalBeforeEach.spec.js]: use camelCase only

File: spec/all.globalBeforeEach.spec.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+{
+  browser.setValue('#loginEmail', 'sroy@sapient.com');
+  browser.setValue('#loginPassword', 'welcome1');
+}
+
+function isMerge()
+{
+  var el = browser.getTitle();
+  if(el == 'My Basket')
```
Comment (SunilSyal) [inline spec/all.globalBeforeEach.spec.js]: No if/else in test case.

File: spec/all.globalBeforeEach.spec.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  }
+  else {
+  browser.setValue('.collection--store-finder__search-form__input', 'Covent Garden');
+  }
+}
+
+
+it.only('Place an UK CFTO order with registered user with added item', function() {
+
+    this.timeout(0);
```
Comment (SunilSyal) [inline spec/all.globalBeforeEach.spec.js]: why timeout?

File: spec/all.globalBeforeEach.spec.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    browser.waitForVisible('script + .minibag-container', 10000);
+    browser.click('script + .minibag-container .minibag-summary__button');
+    browser.click('button[data-analyticsfunction="analyticsCheckoutButtonClicked"]');
+    log_in();
+    browser.waitForEnabled('.my-account__subcontent__signIn-btn', 10000);
+    browser.click('.my-account__subcontent__signIn-btn');
+    isMerge();
+    browser.click('.collection--store-finder__search-form__btn');
+    browser.click('.btn--primary');
+    browser.click('.btn--primary');
```
Comment (SunilSyal) [inline spec/all.globalBeforeEach.spec.js]: Add inline comments about these statements

File: spec/all.globalBeforeEach.spec.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    browser.setValue('#expiryYear', '18');
+    browser.setValue('div[ng-form="paymentDetailsCtrl.form"] #cvv', '609');
+    browser.click('button[data-element="AddCardAddressOverlay"]');
+    browser.waitForVisible('#address-details__add', 10000);
+    browser.click('#address-details__add');
+    browser.waitForVisible('.payment-button', 10000);
+    browser.setValue('.credit-debit-card-form #cvv', '609');
+    browser.click('.payment-button');
+    browser.click('input[type="submit"]');
+    expect(browser.element('span[ng-bind="contentText.orderNumber"]').getText()).to.contain('401');
```
Comment (SunilSyal) [inline spec/all.globalBeforeEach.spec.js]: remove angular bindings

File: spec/all.globalBeforeEach.spec.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      browser.setValue('.billing-form #address1', '23, Egremont House');
+      browser.setValue('.billing-form #city', 'London');
+      browser.setValue('.billing-form #zipCode', 'E20 1BF');
+      browser.setValue('#guestemail', 'sroy@sapient.com');
+      browser.waitForEnabled('.payment-button ', 30000);
+      browser.click('.payment-button');
+      browser.click('.order-review__terms .lbl-checkbox');
+      browser.waitForEnabled('input[type="submit"]', 5000);
+      browser.click('input[type="submit"]');
+      expect(browser.element('span[ng-bind="contentText.orderNumber"]').getText()).to.contain('401');
```
Comment (SunilSyal) [inline spec/all.globalBeforeEach.spec.js]: Same issue

File: spec/all.globalBeforeEach.spec.js (hunk 1 above)
Comment (sumitroy2011) [reply spec/all.globalBeforeEach.spec.js]: Done. all the function naming convention has been corrected

File: spec/all.globalBeforeEach.spec.js (hunk 2 above)
Comment (sumitroy2011) [reply spec/all.globalBeforeEach.spec.js]: Removed the If/Else from functions

File: spec/all.globalBeforeEach.spec.js (hunk 3 above)
Comment (sumitroy2011) [reply spec/all.globalBeforeEach.spec.js]: As discussed, This is needed as some of the JS functions needs timeout to be predefined. 
Either those can be handled using promises or completely nullifying the timeout by using this function

File: spec/all.globalBeforeEach.spec.js (hunk 4 above)
Comment (sumitroy2011) [reply spec/all.globalBeforeEach.spec.js]: I have commented all the functions & steps and transformed the redundant code into functions

File: spec/all.globalBeforeEach.spec.js (hunk 6 above)
Comment (sumitroy2011) [reply spec/all.globalBeforeEach.spec.js]: Removed all angular functions

**Comment by sumitroy2011** [comment]
Amended all the today's comments

**Pull request merged**